External Renderer/Assets/Scripts/IP Transmission/Receiver.cs

```diff
@@ -36,53 +37,63 @@ public Receiver(int port = 11000)
             }
             catch (Exception e) // remove pokemon exception handle
             {
-                Debug.LogError(e.ToString());
+                Debug.Log(e.ToString());
             }
         }
 
-        public void RecieveMessage(Action<string> dataReceivedCallback)
+        public async void ReceiveMessage(Action<string> dataReceivedCallback)
         {
-            // byte cache for what is recieved
-            byte[] bytes = new byte[1024];
-            using (MemoryStream cache = new MemoryStream(1000))
+            string data = "";
+            while (true)
             {
-                try
-                {
-                    Socket handler = _listener.Accept();
-
-                    // Incoming data from the client.
-                    while (true)
+                data =
+                    await Task.Run(() =>
                     {
-                        int bytesReceived = handler.Receive(bytes);
-                        cache.Write(bytes, 0, bytesReceived);
-
-                        // check if the client disconnected
-                        if (handler.Poll(1, SelectMode.SelectRead) && handler.Available == 0)
+                        // byte cache for what is recieved
+                        byte[] bytes = new byte[1024];
+                        using (MemoryStream cache = new MemoryStream())
                         {
-                            break;
-                        }
-                    }
+                            try
+                            {
+                                Socket handler = _listener.Accept();
 
-                    cache.Seek(0, SeekOrigin.Begin);
+                                // Incoming data from the client.
+                                while (true)
```

Comment:
I don't have a strong personal opinion, but many people recommend against while(true) if possible, as general practice. This loop seems easy to change, by setting the negation of the "if' condition as the while condition. (You may have to consider if that requires a "do while" loop, I haven't thought about it)